crates/matrix-sdk/src/event_cache/mod.rs

```diff
@@ -551,10 +593,21 @@ impl EventCacheInner {
                     pagination_status,
                     room_id.to_owned(),
                     auto_shrink_sender,
+                    self.generic_room_event_cache_update_sender.clone(),
                 );
 
                 by_room_guard.insert(room_id.to_owned(), room_event_cache.clone());
 
+                // If at least one event has been loaded, it means there is a timeline. Let's
+                // emit a generic update.
+                if has_at_least_one_event {
```

Comment:
So the creation of an empty timeline is not something we want to hear about?

Can timelines _become_ empty through redactions?

Reply:
> So the creation of an empty timeline is not something we want to hear about?

We may, at some point, but so far only `RoomEventCacheGenericUpdate::TimelineUpdated` exists. If we are interested by another event, we can change the logic here.

> Can timelines become empty through redactions?

No, because it will contain at least 2 events: one for the original event, and one for the redaction event. In all cases, this is a _timeline update_.